### DISCUSSION TITLE
Intercept and reject requests based on body content

### DISCUSSION THREAD

**Discussion by ortonomy** [chimurai/http-proxy-middleware#540] [Q&A]
I want to be able to intercept and reject a request through the proxy by examining body content.

For example, my request coming through is a grapql query or mutation in ``body`` of ``POST``

````
mutation {
  registerUser ( ... ) {
    ...
  }
}
````

and I want to be able to examine the content body for ``registerUser`` (or something else) and reject with a ``403 - Forbidden``. 

I feel it's similar, but the opposite to here: https://github.com/chimurai/http-proxy-middleware/issues/97

So I'd like to use ``onProxyReq``, but I dont' know how to read the body and then make an action using ``req`` / ``res``.  Anyone can help?

**Comment by stevenkissack** [comment]
Did you ever resolve this? I am trying to do the exact same. I also want to modify the graphql query in certain cases, and reject all mutations

**Reply by nsrCodes** [reply]
You can do this with [Requestly](https://requestly.io/downloads/) by creating a Modify Response Rule withthe Request Payload filter. Link to article - https://requestly.io/blog/mocking-graphql-api-response/

**Comment by chimurai** [comment]
You might want to try using a library like [`nock`](https://www.npmjs.com/package/nock).
It has tons of options to control the response.

The following example will just respond with 403 on every graphql request.
```js
var scope = nock('http://my.domain.com')
  .intercept('/graphql', 'POST')
  .reply(403);
```

Check their documentation for more fine grained control.

Please let us know if you have any succes with `nock`.
